### ISSUE TITLE
simplify works better than trigsimp

### ISSUE THREAD

**Issue by meganly** [sympy/sympy#17778]
`simplify` does better than `trigsimp` in simplifying the following expression.

```
>>> from sympy import trigsimp, simplify, symbols, cos, sin, cot
>>> x = symbols('x')
>>> expr = 1/(sin(x)*cos(x))-cot(x)
>>> simplify(expr)
tan(x)
>>> trigsimp(expr)
-cot(x) + 2/sin(2*x)
```

As @asmeurer suspected, simplify writes the expression as a combined fraction first
```
>>> trigsimp(cancel(expr))
tan(x)
```

It seems that trigsimp should try cancel() first somewhere in here
https://github.com/sympy/sympy/blob/fed3bb83dec834bd75fd8bcd68fc0c31387f394a/sympy/simplify/trigsimp.py#L1132-L1137

**Comment by asmeurer** [comment]
This would also be a good use case for the groebner method, but it doesn't seem to know about cot.

**Comment by meganly** [comment]
True, but the default trigsimp algorithm doesn't call the groebner method.

Would adding `lambda x: _eapply(cancel, x, trigs),` to the tree shown in the code above make trigsimp also try canceling?  I'm a little confused how the code to construct the tree works.

**Comment by asmeurer** [comment]
`greedy` tries the functions greedily to minimize the objective function which is defined above that code.

I think a problem that you would run into is that the objective is the number of trig functions (`L`) and then the total number of operations. But cancel actually increases the number of operations

```
>>> expr = 1/(sin(x)*cos(x))-cot(x)
>>> expr.count_ops()
6
>>> cancel(expr).count_ops()
11
```

So my guess from reading the code is that it wouldn't work. But I also didn't try it, so I could be wrong.

**Comment by asmeurer** [comment]
`cancel` also increases the number of trig functions

```
>>> from sympy.simplify.fu import L
>>> L(expr)
3
>>> L(cancel(expr))
5
```
